certora/specs/GuardianStorage.spec

```diff
@@ -0,0 +1,372 @@
+/*
+ * Spec for linked list reachability for Guardians.
+ * This file is derived from OwnerReach.spec for a Safe account.
+ * https://github.com/safe-global/safe-smart-account/blob/main/certora/specs/OwnerReach.spec
+ *
+ * This file uses a reach predicate:
+ *    ghost reach(address, address, address) returns bool
+ * to represent the transitive of the next
+ * relation given by the "owners" field relation for a specific wallet.
+ *
+ * The idea comes from the paper
+ *
+ * [1] Itzhaky, S., Banerjee, A., Immerman, N., Nanevski, A., Sagiv, M. (2013).
+ *     Effectively-Propositional Reasoning about Reachability in Linked Data Structures.
+ *     In: CAV 2013. Springer, https://doi.org/10.1007/978-3-642-39799-8_53
+ */
+
+using SafeHarness as safeContract;
+using SocialRecoveryModule as socialRecoveryModule;
+
+methods {
+    // Safe Functions
+    function safeContract.isModuleEnabled(address) external returns (bool) envfree;
+
+    // GuardianManager Functions
+    function threshold(address) external returns (uint256) envfree;
+    function isGuardian(address, address) external returns (bool) envfree;
+
+}
+
+persistent ghost reach(address, address, address) returns bool {
```

Comment:
nit: Can we use consistent naming for the ghost variables? We have `ghost*` in other places and would prefer to keep that naming here as well.

Typically, you could argue that this is tooling's responsibility, but I don't think there is sufficient tooling for CVL source.

Reply:
https://github.com/5afe/CandideWalletContracts/pull/20/commits/069525f3851c844df8bcc94f38fde5dc3e1d8c6f